src/AppInstallerCommonCore/AppInstallerTelemetry.cpp

```diff
@@ -43,38 +43,38 @@ namespace AppInstaller::Logging
         {
             Telemetry().LogFailure(info);
         }
-
-        GUID CreateGuid()
-        {
-            GUID result{};
-            (void)CoCreateGuid(&result);
-            return result;
-        }
-
-        const GUID* GetActivityId()
-        {
-            static GUID activityId = CreateGuid();
-            return &activityId;
-        }
     }
 
     TelemetryTraceLogger::TelemetryTraceLogger()
     {
-        // TODO: Needs to be made a singleton registration/removal in the future
+        HRESULT hr = S_OK;
+
         RegisterTraceLogging();
 
-        m_isSettingEnabled = !Settings::User().Get<Settings::Setting::TelemetryDisable>();
+        RegisterTraceLogging();
+
+        hr = CoCreateGuid(&m_activityId);
+        if (FAILED(hr))
+        {
+            TraceLoggingWriteActivity(
```

Comment:
I would honestly just remove this event rather than potentially write it when we don't have the setting loaded.

Reply:
Makes Sense. No point in creating a new event. I will just log any logging related failures to text file.